python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -354,15 +354,29 @@ def removeNodeAttribute(self, attrib_name):
         if attrib_name in self._data.keys():
             del self._data[attrib_name]
 
-    def removeNodeIndexCacheAttribute(self, attrib_name):
-        if self._index_cache and attrib_name in self._index_cache.keys():
-            del self._index_cache[attrib_name]
-
     def setNodeIndexCache(self, new_index_cache):
+        if self.fullyLoadedFromDisk():
+            logger.debug("Warning: Setting IndexCache data on live object, please avoid!")
         setattr(self, '_index_cache', new_index_cache)
 
-    def getNodeIndexCache(self):
+    def getNodeIndexCache(self, force_cache=False):
```

Comment:
No, I think it makes the most sense to add them as we go. No one is going to sit down and take the time to understand the inner working of every function simply to write a docstring, however if someone is working on the function then they best understand it and writing a docstring at that time is most useful.

On top of this, we should not be adding any *new* arguments to functions or adding new functions without docstrings.